go/libraries/doltcore/sqle/statspro/dolt_stats.go

```diff
@@ -26,54 +26,158 @@ import (
 	Chunks []hash.Hash
 	// Active maps a chunk/bucket address to its position in
 	// the histogram. 1-indexed to differentiate from an empty
 	// field on disk
 	Active map[hash.Hash]int
+	Hist   sql.Histogram
+}
+
+var _ sql.Statistic = (*DoltStats)(nil)
+
+func (s *DoltStats) WithColSet(set sql.ColSet) sql.Statistic {
+	ret := *s
+	ret.Statistic = ret.Statistic.WithColSet(set).(*stats.Statistic)
+	return &ret
+}
+
+func (s *DoltStats) WithFuncDeps(set *sql.FuncDepSet) sql.Statistic {
+	ret := *s
+	ret.Statistic = ret.Statistic.WithFuncDeps(set).(*stats.Statistic)
+	return &ret
+}
+
+func (s *DoltStats) WithDistinctCount(u uint64) sql.Statistic {
+	ret := *s
+	ret.Statistic = ret.Statistic.WithDistinctCount(u).(*stats.Statistic)
+	return &ret
+}
+
+func (s *DoltStats) WithRowCount(u uint64) sql.Statistic {
+	ret := *s
+	ret.Statistic = ret.Statistic.WithRowCount(u).(*stats.Statistic)
+	return &ret
+}
+
+func (s *DoltStats) WithNullCount(u uint64) sql.Statistic {
+	ret := *s
+	ret.Statistic = ret.Statistic.WithNullCount(u).(*stats.Statistic)
+	return &ret
+}
+
+func (s *DoltStats) WithAvgSize(u uint64) sql.Statistic {
+	ret := *s
+	ret.Statistic = ret.Statistic.WithAvgSize(u).(*stats.Statistic)
+	return &ret
+}
+
+func (s *DoltStats) WithLowerBound(row sql.Row) sql.Statistic {
+	ret := *s
+	ret.Statistic = ret.Statistic.WithLowerBound(row).(*stats.Statistic)
+	return &ret
+}
+
+func (s *DoltStats) RowCount() uint64 {
+	return s.Statistic.RowCount()
+}
 
-	RowCount      uint64
-	DistinctCount uint64
-	NullCount     uint64
-	AvgSize       uint64
-	Qual          sql.StatQualifier
-	CreatedAt     time.Time
-	Histogram     DoltHistogram
-	Columns       []string
-	Types         []sql.Type
-	IdxClass      uint8
-	LowerBound    sql.Row
-	Fds           *sql.FuncDepSet
-	ColSet        sql.ColSet
+func (s *DoltStats) DistinctCount() uint64 {
+	return s.Statistic.DistinctCount()
+}
+
+func (s *DoltStats) NullCount() uint64 {
+	return s.Statistic.NullCount()
+
+}
+
+func (s *DoltStats) AvgSize() uint64 {
+	return s.Statistic.AvgSize()
+
+}
+
+func (s *DoltStats) CreatedAt() time.Time {
+	return s.Statistic.CreatedAt()
+
+}
+
+func (s *DoltStats) Columns() []string {
+	return s.Statistic.Columns()
+}
+
+func (s *DoltStats) Types() []sql.Type {
+	return s.Statistic.Types()
+}
+
+func (s *DoltStats) Qualifier() sql.StatQualifier {
+	return s.Statistic.Qualifier()
+}
+
+func (s *DoltStats) IndexClass() sql.IndexClass {
+	return s.Statistic.IndexClass()
+}
+
+func (s *DoltStats) FuncDeps() *sql.FuncDepSet {
+	return s.Statistic.FuncDeps()
+}
+
+func (s *DoltStats) ColSet() sql.ColSet {
+	return s.Statistic.ColSet()
+}
+
+func (s *DoltStats) LowerBound() sql.Row {
+	return s.Statistic.LowerBound()
 }
 
 func NewDoltStats() *DoltStats {
-	return &DoltStats{mu: &sync.Mutex{}, Active: make(map[hash.Hash]int)}
+	return &DoltStats{mu: &sync.Mutex{}, Active: make(map[hash.Hash]int), Statistic: &stats.Statistic{}}
+}
+
+func (s *DoltStats) ToInterface() interface{} {
```

Comment:
This should probably be called `JsonMap` or similar and potentially have the return type `map[string]any`

Reply:
i think it can return a list of interfaces also, `[]interface{}`